tests/core/flow/flow_benchmark_test.go

```diff
@@ -0,0 +1,114 @@
+package flow
+
+import (
+	"testing"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/core/flow"
+	"github.com/alibaba/sentinel-golang/core/stat"
+)
+
+var (
+	slot *flow.Slot
+	ctx  *base.EntryContext
+)
+
+func init() {
+	slot = flow.DefaultSlot
+	res := base.NewResourceWrapper("abc", base.ResTypeCommon, base.Inbound)
+	resNode := stat.GetOrCreateResourceNode("abc", base.ResTypeCommon)
+	ctx = &base.EntryContext{
+		Resource: res,
+		StatNode: resNode,
+		Input: &base.SentinelInput{
+			BatchCount: 1,
+		},
+		RuleCheckResult: nil,
+		Data:            nil,
+	}
+}
+
+func loadDirectRejectRule() {
+	r1 := &flow.Rule{
+		Resource:               "abc",
+		TokenCalculateStrategy: flow.Direct,
+		ControlBehavior:        flow.Reject,
+		Threshold:              10,
+		StatIntervalInMs:       1000,
+		RelationStrategy:       flow.CurrentResource,
+	}
+	flow.LoadRules([]*flow.Rule{r1})
+}
+
+func loadWarmUpRejectRule() {
+	r1 := &flow.Rule{
+		Resource:               "abc",
+		TokenCalculateStrategy: flow.WarmUp,
+		ControlBehavior:        flow.Reject,
+		Threshold:              100,
+		WarmUpPeriodSec:        10,
+		WarmUpColdFactor:       3,
+		StatIntervalInMs:       1000,
+	}
+	flow.LoadRules([]*flow.Rule{r1})
+}
+
+func Benchmark_DirectReject_SlotCheck_4(b *testing.B) {
+	loadDirectRejectRule()
+	b.ReportAllocs()
+	b.ResetTimer()
+	b.SetParallelism(4)
+	for i := 0; i < b.N; i++ {
+		slot.Check(ctx)
+	}
+}
```

Comment:
Is it like this?
```go
b.RunParallel(func(pb *testing.PB) {
		for pb.Next() {
			slot.Check(ctx)
		}
	})
```

Reply:
Running in parallel might be better for comparing different check implementations, but for now I just need to compare the results of running with different Numbers of P

Reply:
good